server/src/main/java/com/objectcomputing/checkins/services/checkin_notes/CheckinNoteController.java

```diff
@@ -45,61 +55,119 @@ public CheckinNoteController(CheckinNoteServices checkinNoteServices) {
      * @param request
-     * @return
+     * @return {@link HttpResponse <CheckinNote>}
      */
     @Post("/")
     @Secured({RoleType.Constants.PDL_ROLE, RoleType.Constants.ADMIN_ROLE})
-    public HttpResponse<CheckinNote> createCheckinNote(@Body @Valid CheckinNoteCreateDTO checkinNote, HttpRequest<CheckinNoteCreateDTO> request) {
-        CheckinNote newCheckinNote = checkinNoteServices.save(new CheckinNote(checkinNote.getCheckinid(), checkinNote.getCreatedbyid()
-                , checkinNote.getDescription()));
-        return HttpResponse.created(newCheckinNote)
-                .headers(headers -> headers.location(URI.create(String.format("%s/%s", request.getPath(), newCheckinNote.getId()))));
-
+    public Single<HttpResponse<CheckinNote>> createCheckinNote(@Body @Valid CheckinNoteCreateDTO checkinNote,
+                                                                    HttpRequest<CheckinNoteCreateDTO> request) {
+        return Single.fromCallable(() -> checkinNoteServices.save(new CheckinNote(checkinNote.getCheckinid(),
+                checkinNote.getCreatedbyid(), checkinNote.getDescription())))
+                .observeOn(Schedulers.from(eventLoopGroup))
+                .map(createdCheckinNote -> {
+                    //Using code block rather than lambda so we can log what thread we're in
+                    return (HttpResponse<CheckinNote>) HttpResponse
+                    .created(createdCheckinNote)
+                    .headers(headers -> headers.location(
+                        URI.create(String.format("%s/%s", request.getPath(), createdCheckinNote.getId()))));
+                }).subscribeOn(Schedulers.from(ioExecutorService));
     }
 
-    /**
+/**
      * Update a check in note
      *
-     * @param checkinNote
-     * @param request
-     * @return
+     * @param checkinNote, {@link CheckinNote}
+     * @return {@link HttpResponse< CheckinNote >}
      */
     @Put("/")
     @Secured({RoleType.Constants.PDL_ROLE, RoleType.Constants.ADMIN_ROLE})
-    public HttpResponse<CheckinNote> updateCheckinNote(@Body @Valid CheckinNote checkinNote, HttpRequest<CheckinNoteCreateDTO> request) {
-        CheckinNote updateCheckinNote = checkinNoteServices.update(checkinNote);
-        return HttpResponse.ok().headers(headers -> headers.location(
-                URI.create(String.format("%s/%s", request.getPath(), updateCheckinNote.getId()))))
-                .body(updateCheckinNote);
+    public Single<HttpResponse<CheckinNote>> updateCheckinNote(@Body @Valid CheckinNote checkinNote,
+                                            HttpRequest<CheckinNote> request) {
+        if (checkinNote == null) {
+            return Single.just(HttpResponse.ok());
+        }
+        return Single.fromCallable(() -> checkinNoteServices.update(checkinNote))
+            .observeOn(Schedulers.from(eventLoopGroup))
+            .map(updatedCheckinNote -> //This lambda expression is the preferred way to do this kind of simple mapping.
```

Comment:
The lambda will look a lot like this. This comment was also only there to be demonstrative.

Reply:
Updated